tests/thread_float/tests/01-run.py

```diff
@@ -28,9 +28,21 @@ def same_computation_as_in_c_prog(thread_num):
 
 def testfunc(child):
     child.expect_exact("THREADS CREATED")
-    child.expect_exact("THREAD t1 start")
-    child.expect_exact("THREAD t2 start")
-    child.expect_exact("THREAD t3 start")
+    threads_started = set()
+    child.expect(r"THREAD t(\d) start\r\n")
+    thread_num = int(child.match.group(1))
+    assert thread_num in {1, 2, 3}, "thread number has to be valid"
+    threads_started.add(thread_num)
+    child.expect(r"THREAD t(\d) start\r\n")
+    thread_num = int(child.match.group(1))
+    assert thread_num in {1, 2, 3}, "thread number has to be valid"
+    assert thread_num not in threads_started, "same thread must not be launched twice"
+    threads_started.add(thread_num)
+    child.expect(r"THREAD t(\d) start\r\n")
+    thread_num = int(child.match.group(1))
+    assert thread_num in {1, 2, 3}, "thread number has to be valid"
+    assert thread_num not in threads_started, "same thread must not be launched twice"
+    threads_started.add(thread_num)
```

Comment:
Some code deduplication would be nice:
```suggestion
    for _ in range(3):
        child.expect(r"THREAD t(\d) start\r\n")
        thread_num = int(child.match.group(1))
        assert thread_num in {1, 2, 3}, "thread number has to be valid"
        assert thread_num not in threads_started, "same thread must not be launched twice"
        threads_started.add(thread_num)
```